### DISCUSSION TITLE
Documentation about Genevi Extension: Service IDs 0xF01, 0xF02, 0xF03 and 0xF04

### DISCUSSION THREAD

**Discussion by NeodaTech** [COVESA/dlt-viewer#231] [General]
Hello,

i don't know if this is the right place to talk about it but i currently don't have any idea for another Place, since i have a question but also tell you something about my use cases and if you have any other idea how i could use the DLT-Viewer for my use cases i would be also very happy to hear about that :D

I am currently working (finalized a certain state that is working) on a Python implementation to connect via a Socket to a Target and to analyze the Stability and Decode the DLT Steam that is received according to the DLT Specification. Those Decoded DLT Message Objects are then stored in an Array, wich then can be accessed and further processed by the user or/and written to a logfile (.dlt format as specified or .txt Format where a DLT Message is display in a string representation).
Later i also want send customized DLT Messages (Control Messages) to a Target for Testing purposes and of course also decode the Control Messages (in progress).
I saw, that GENEVI also Extended the DLT Protocol, especially the DLT Control Messages (https://at.projects.genivi.org/wiki/display/PROJ/GENIVI+DLT+Protocol+Extensions) with the Service IDs 0xF01, 0xF02, 0xF03 and 0xF04. **Is there any kind of documentation how the DLT-Viewer is decoding/encoding the payload for those additional Service IDs or do i have to check the implementation in the Code ?**

why do i need this ?

I require Realtime access to DLT Messages and the possibility to react on certain DLT Messages inside another (Python Based) Testing-Framework. This is not Possible (according to what i know) with the DLT Viewer, since with the DLT Viewer you can just start and stop the recording for a defined period of time and then i would need to process the generated Logfile in a sperate step.
I did not find any way how i could use DLT Viewer in a way that it for examples waits for one or more specific DLT Message and then gives me confirmation that this messages are received or even extract something via a regular Expression from the payload that dynamically changes according to the Test-Input i give for example via CAN. 
This is especially useful if you for example don't know if a message comes after 1 Second or 5 Minutes. you just want to wait (with a timeout of 5 minutes) until the Message is received and directly continue with the Test Execution and not wait any additional/more time then necessary, since those 5 minutes are just the worst-case scenario.

**Comment by alexmucde** [comment]
@NeodaTech Please consider the source code as doumentation. There is no extra documentation available beside the AUTOSAR specification of DLT.
Regarding forwarding message and test adapter please consider the lattest DLTTestRobot plugin.
